libs/storage/Tsavorite/cs/src/core/Index/Tsavorite/Tsavorite.cs

```diff
@@ -653,32 +653,87 @@ private unsafe string DumpDistributionInternal(int version)
                         x.word = b.bucket_entries[bucket_entry];
+
+                        total_entries_with_tentative_bit_set += x.Tentative ? 1 : 0;
+
                         if (((!x.ReadCache) && (x.Address >= beginAddress)) || (x.ReadCache && (x.AbsoluteAddress >= readCacheBase.HeadAddress)))
                         {
                             if (tags.Contains(x.Tag) && !x.Tentative)
                                 throw new TsavoriteException("Duplicate tag found in index");
                             tags.Add(x.Tag);
-                            ++cnt;
+                            ++total_valid_records_in_this_bucket_cnt;
                             ++total_record_count;
                         }
+                        else if (x.word != default && x.Address < beginAddress)
+                        {
+                            if (is_bucket_in_ofb_table)
+                                ++total_entries_in_ofb_below_begin_address;
+
+                            ++total_entries_below_begin_address;
+                        }
+                        else
+                        {
+                            ++zeroed_out_slots;
+                        }
+
+                        if (is_bucket_in_ofb_table)
+                            total_valid_entries_in_ofb_in_this_bucket_cnt++;
                     }
-                    if ((b.bucket_entries[Constants.kOverflowBucketIndex] & Constants.kAddressMask) == 0) break;
+
+                    total_zeroed_out_slots += zeroed_out_slots;
+
+                    if (is_bucket_in_ofb_table)
+                    {
+                        if (!slots_unused_by_buckets_ofb_histogram.ContainsKey(zeroed_out_slots))
+                            slots_unused_by_buckets_ofb_histogram[zeroed_out_slots] = 0;
+                        slots_unused_by_buckets_ofb_histogram[zeroed_out_slots]++;
+                    }
+                    else
+                    {
+                        if (!slots_unused_by_nonofb_buckets_histogram.ContainsKey(zeroed_out_slots))
+                            slots_unused_by_nonofb_buckets_histogram[zeroed_out_slots] = 0;
+                        slots_unused_by_nonofb_buckets_histogram[zeroed_out_slots]++;
+                    }
+
+                    if ((b.bucket_entries[Constants.kOverflowBucketIndex] & Constants.kAddressMask) == 0)
+                        break;
+
                     b = *(HashBucket*)overflowBucketsAllocator.GetPhysicalAddress(b.bucket_entries[Constants.kOverflowBucketIndex] & Constants.kAddressMask);
+                    is_bucket_in_ofb_table = true;
                 }
 
-                if (!histogram.ContainsKey(cnt)) histogram[cnt] = 0;
-                histogram[cnt]++;
+                if (!histogram.ContainsKey(total_valid_records_in_this_bucket_cnt)) histogram[total_valid_records_in_this_bucket_cnt] = 0;
+                histogram[total_valid_records_in_this_bucket_cnt]++;
```

Comment:
it doesn't matter so much here, but generally this should be "set to 1 else increment" to avoid an additional lookup